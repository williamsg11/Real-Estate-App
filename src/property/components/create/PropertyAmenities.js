/**
 * @flow
 */
import React, {PropTypes} from 'react';
import {
  ListView,
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import colors from '../../../common/colors';
import Separator from './../../../components/Separator';

export default class PropertyAmenities extends React.Component {
  static propTypes = {
    selected: PropTypes.array.isRequired,
    updateListing: PropTypes.func.isRequired,
    collection: PropTypes.array.isRequired,
  };

  state = {
    descriptionHeight: 40,
    disabled: true,
  };

  renderRow = item => {
    const {updateListing, selected} = this.props;
    return (
      <TouchableHighlight
        style={{flex: 1}}
        onPress={() => updateListing(item)}
        underlayColor="transparent"
        key={item}>
        <View style={styles.row}>
          <Text style={styles.title}>{item}</Text>
          <View style={styles.checkbox}>
            {selected &&
              selected.includes(item) &&
              <FontAwesome
                key={item}
                name="check"
                size={16}
                color={colors.green}
              />}
          </View>
        </View>
      </TouchableHighlight>
    );
  };

  render() {
    const {collection, header, footer} = this.props;

    let ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 != r2});
    let dataSource = ds.cloneWithRows(collection);

    return (
      <View style={styles.container}>
        <View style={styles.menuContainer}>
          {header}
        </View>
        <View style={{flex: 5}}>
          <ListView
            dataSource={dataSource}
            style={styles.list}
            enableEmptySections={true}
            renderRow={this.renderRow}
            automaticallyAdjustContentInsets={false}
            showsVerticalScrollIndicator={false}
            contentInset={{bottom: 50}}
            renderHeader={() => <View />}
            renderSeparator={(sectionId, rowId) => (
              <Separator style={{marginVertical: 10}} key={rowId} />
            )}
          />
        </View>
        {footer}
      </View>
    );
  }
}

PropertyAmenities.prototypes = {
  collection: React.PropTypes.array.isRequired,
  updateListing: React.PropTypes.func.isRequired,
  selected: React.PropTypes.array.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  menuContainer: {
    flex: 1,
    backgroundColor: colors.smokeGreyLight,
    paddingHorizontal: 10,
    paddingVertical: 30,
  },
  list: {
    flex: 1,
    padding: 10,
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  title: {
    flex: 1,
    fontWeight: '100',
    color: colors.darkGrey,
  },
  checkbox: {
    width: 24,
    height: 24,
    borderRadius: 12,
    borderColor: colors.smokeGreyDark,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 10,
  },
});
