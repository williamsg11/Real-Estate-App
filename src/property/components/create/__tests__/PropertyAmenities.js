import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PropertyAmenities from './../PropertyAmenities';

test('renders PropertyAmenities', () => {
  const tree = renderer
    .create(
      <PropertyAmenities
        selected={['swimming pool']}
        updateListing={() => {}}
        collection={['Sauna', 'swimming pool']}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
