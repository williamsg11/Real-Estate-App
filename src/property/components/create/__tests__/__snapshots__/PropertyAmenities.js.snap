// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PropertyAmenities 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#e3e7ed",
        "flex": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 30,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 5,
      }
    }
  >
    <RCTScrollView
      alwaysBounceHorizontal={undefined}
      alwaysBounceVertical={true}
      automaticallyAdjustContentInsets={false}
      contentInset={
        Object {
          "bottom": 50,
        }
      }
      dataSource={
        ListViewDataSource {
          "items": 2,
        }
      }
      enableEmptySections={true}
      onContentSizeChange={null}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={undefined}
      onResponderTerminationRequest={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onScrollShouldSetResponder={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      renderHeader={[Function]}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
      renderSeparator={[Function]}
      sendMomentumEvents={false}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "overflow": "scroll",
          },
          Object {
            "flex": 1,
            "padding": 10,
          },
        ]
      }
    >
      <RCTScrollContentView
        collapsable={false}
        removeClippedSubviews={undefined}
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        <View />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flex": 1,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#484848",
                  "flex": 1,
                  "fontWeight": "100",
                }
              }
            >
              Sauna
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#5d5f63",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "height": 24,
                  "justifyContent": "center",
                  "marginHorizontal": 10,
                  "width": 24,
                }
              }
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flex": 1,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#484848",
                  "flex": 1,
                  "fontWeight": "100",
                }
              }
            >
              swimming pool
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#5d5f63",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "height": 24,
                  "justifyContent": "center",
                  "marginHorizontal": 10,
                  "width": 24,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#388704",
                      "fontSize": 16,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                ï€Œ
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollContentView>
    </RCTScrollView>
  </View>
</View>
`;
